File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.java
import java.util.ArrayList;

class Solution {
    public int solution(int storey) {
        int answer = 0;
        String storeyString = String.valueOf(storey);
        int length = storeyString.length();
        ArrayList<Integer> list = new ArrayList<>();
        list.add(0);

        for(int i = 0; i<length; i++){
            char ch = storeyString.charAt(i);
            list.add(Integer.parseInt(String.valueOf(ch)));
        }

        for(int i = length; i > 0 ; i--){
            if(list.get(i) > 5 || (list.get(i) >= 5 && list.get(i-1) >= 5)){
                list.set(i-1, list.get(i-1) + 1);
                answer += 10 - list.get(i);
            }else{
                answer += list.get(i);
            }
        }
        
        answer += list.get(0);

        return answer;
    }
}

File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.js
function solution(storey) {
  var answer = 0;
  var toStr = `${storey}`
  var len = toStr.length;

  var arr = [0];
  for(let i = 0; i<len; i++){
    arr.push(parseInt(toStr[i], 10));
  }

  for(let i = len; i > 0; i--){

    if(arr[i] > 5 || (arr[i] >= 5 && arr[i-1] >= 5)){
      arr[i - 1] += 1;
      answer += 10 - arr[i];
    }else{
      answer += arr[i];
    }
  }

  if(arr[0] > 5){
    answer += 10 - arr[0] + 1;
  }else{
    answer += arr[0];
  }

  return answer;
}